pkg/mic/mic.go

```diff
@@ -261,24 +275,48 @@ func (c *Client) Sync(exit <-chan struct{}) {
 
 				glog.V(5).Infof("Initiating assigned id creation for pod - %s, binding - %s", createID.Spec.Pod, binding.Name)
 
-				err = c.createAssignedIdentityDeps(&createID, id, node)
+				err = c.createAssignedIdentity(&createID)
```

Comment:
Is there a way we can perform this create when we start looking at the node map ? Since we don't track state of the assigned identities, we could have a large time gap between the time the assignment is created and then the node assignment was made. Is that a possibility ? If so, can we mitigate it some how without additional state ?

Reply:
@kkmsft This mimics the current logic. Currently, we create the assigned identity, then make the request for node assignment. The timegap still remains the same with this change because we are batching all the assignments and making the single call. So the time for assignment still remains the same. There shouldn't be a long time gap here.